### PR TITLE
Issue 22520 copy site does not copy all content when a site has more than 10000 contentlets

### LINKED ISSUE #22520: Copy Site does not copy all content when a site has more than 10000 contentlets
**Describe the bug**
Copy site does not copy all content when a site has more than 10000 contentlets

Reproduced on: 5.3.8.10/22.06

Related ticket: https://dotcms.zendesk.com/agent/tickets/106874

**To Reproduce**
From a site with more than 10000 total contentlets
1. Attempt to copy the site
2. It will complete successfully, however not all content will have been copied

**Expected behavior**
Copy site should copy all content. Currently it is hitting the limit for our Elasticsearch queries: 

https://github.com/dotCMS/core/blob/88aef64a9b92923f1c2089af9ceed262198a8f9e/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java#L178-L178

This is because the search we do to pull in a list of content to copy is this:
`+conHost:{HostIdentifier} +working:true`
We pass a limit of -1, which gets converted to the MAX_LIMIT of 10000, so only 10k contentlets are pulled in. This is related to a limit set on our indexes `index.max_result_window` which is capped at 10k. 

In order to correct this we would need to both up the limit and the max_result_window, so it makes more sense to paginate the query.

We do have the ability to use the scroll API to paginate the query: 
https://github.com/dotCMS/core/blob/2e80d1dd0c19c6499ca165979cac8cf307c6098f/dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java#L1097

The simple solution might be setting the limit sent by copy site to something much larger, as sending -1 gets it set to the MAX_LIMIT. Perhaps make it a configurable limit.

### PR DESCRIPTION
**Pull request by freddyucv**
When a Suer try to copy a site with more than 10k of contentlets it just copy 10k of them because ES have a LIMIT, so now we are paginated the contetlets where the site is copied.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/util/contentlet/pagination/ContentletsPaginatedIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,332 @@
+package com.dotmarketing.util.contentlet.pagination;
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/util/contentlet/pagination/ContentletsPaginatedIntegrationTest.java:1]: This test needs to be included in MainSuite.java

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java [hunk 2]
```diff
@@ -262,7 +264,10 @@ default Contentlet move(final Contentlet contentlet, final User user, final Host
 	 */
 	public List<Contentlet> findContentletsByHost(Host parentHost, List<Integer> includingContentTypes, List<Integer> excludingContentTypes, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException;
 
+	public ContentletsPaginated findContentletsPaginatedByHost(Host parentHost, List<Integer> includingContentTypes, List<Integer> excludingContentTypes, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException;
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java:267]: Missing Javadoc.

File: dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletPaginatedBuilder.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.UtilMethods;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import io.vavr.Lazy;
+
+public class ContentletPaginatedBuilder {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletPaginatedBuilder.java:11]: Missing Javadoc.

File: dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletsPaginated.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PaginatedArrayList;;
+import com.liferay.portal.model.User;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import org.jetbrains.annotations.NotNull;
+
+public class ContentletsPaginated implements Iterable<Contentlet> {
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletsPaginated.java:18]: Missing Javadoc.

File: dotCMS/src/integration-test/java/com/dotmarketing/util/contentlet/pagination/ContentletsPaginatedIntegrationTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/util/contentlet/pagination/ContentletsPaginatedIntegrationTest.java:1]: done https://github.com/dotCMS/core/pull/22733/commits/e898cbc868542144ffb120d929883cb21f9e754b#diff-8c127e4dadcd12d480a62cfc2a0dd406dbcb61bab2081608e05dfd4be0df5499R564

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/ContentletAPI.java]: https://github.com/dotCMS/core/pull/22733/commits/e898cbc868542144ffb120d929883cb21f9e754b#diff-fb823c27ba05099d8593a6900c88a586e9e4ceecdf253f3fafc2143d670bc1bcR250

File: dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletsPaginated.java (hunk 4 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletsPaginated.java]: done https://github.com/dotCMS/core/pull/22733/commits/e898cbc868542144ffb120d929883cb21f9e754b#diff-fc748f7490c2d10c66a9f63e6e66786e0830f6f75135a92393958e9d7487c391R17

File: dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletPaginatedBuilder.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/util/contentet/pagination/ContentletPaginatedBuilder.java]: done https://github.com/dotCMS/core/pull/22733/commits/e898cbc868542144ffb120d929883cb21f9e754b#diff-fb1c953d96b3d71b35aafe1e29bf7babf7f3083ca22a0c2e143288eee019df3dR11

**Review by jcastro-dotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**